src/components/Modal/ConnectWalletModal/index.tsx

```diff
@@ -103,6 +114,34 @@ function ConnectWalletModal(props: ReactModal.Props) {
           } as WalletButtonProps,
         ]
       : []),
+    ...wm.wallets.map(
+      ({ info: { logo, windowKey, name, prettyName, downloads } }) => {
+        const isInstalled =
+          windowKey &&
+          (window as unknown as Record<string, unknown>)[windowKey];
+        return {
+          label: prettyName,
+          iconSrc: logo,
+          isInstalled,
+          onClick: async (event) => {
+            try {
+              if (!isInstalled) {
+                const url = downloads?.find(
+                  (d) => d.browser === browserType,
+                )?.link;
+                if (url) window.open(url);
+              }
+              await wm.connect(name, chainName);
+              if (props.onRequestClose) {
+                props.onRequestClose(event);
+              }
+            } catch (error) {
+              console.log(error);
+            }
+          },
+        } as WalletButtonProps;
```

Comment:
```suggestion
        } satisfies WalletButtonProps;
```

nit; this would be better for preventing concealing type errors.

Reply:
Since 'as' is already used throughout this file and the project, I'll consider a broader refactoring in the future

Reply:
I thought we could apply this gradually starting from this line. it's just a nit though - I don't want this to be a blocker.